src/ApiService/ApiService/onefuzzlib/AutoScale.cs

```diff
@@ -77,6 +89,25 @@ public AutoScaleOperations(ILogTracer log, IOnefuzzContext context)
         }
     }
 
+    public async Async.Task<OneFuzzResult<AutoscaleProfile>> GetAutoScaleProfile(Guid scalesetId) {
+        _logTracer.Info($"getting scaleset for existing auto-scale resources {scalesetId}");
+        var settings = _context.Creds.GetResourceGroupResource().GetAutoscaleSettings();
+        if (settings is null) {
+            return OneFuzzResult<AutoscaleProfile>.Error(ErrorCode.INVALID_CONFIGURATION, $"could not find any auto-scale settings for the resource group");
+        } else {
+            await foreach (var setting in settings.GetAllAsync()) {
+
+                if (setting.Data.TargetResourceId.EndsWith(scalesetId.ToString())) {
+                    if (setting.Data.Profiles.Count != 1) {
+                        return OneFuzzResult<AutoscaleProfile>.Error(ErrorCode.INVALID_CONFIGURATION, $"found {setting.Data.Profiles.Count} auto-scale profiles for {scalesetId}");
```

Comment:
In the python code, it's not an error for there to be more than 1 auto scale profile
https://github.com/microsoft/onefuzz/blob/3f86fc8689c5b1785ce9048bc6fded629dbb68d0/src/api-service/__app__/onefuzzlib/azure/auto_scale.py#L245-L253

Reply:
Python will throw if there are 0 profiles though...

Reply:
updated C# code to address the comment